test/verify/machineslib.py

```diff
@@ -1274,6 +1274,63 @@ def installWithErrorTest(dialog):
+        if virtInstallVersion >= "2":
+            self.machine.upload(["verify/files/min-openssl-config.cnf", "verify/files/mock-range-server.py"], "/tmp/")
+
+            # Test detection of ISO file in URL
+            cmds = [
+                # Generate certificate for https server
+                "cd /tmp",
+                "openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -subj '/CN=localhost' -nodes -config /tmp/min-openssl-config.cnf",
+                "cat cert.pem key.pem > server.pem"
+            ]
+            if self.machine.image.startswith("ubuntu") or self.machine.image.startswith("debian"):
+                cmds += [
+                    "cp /tmp/cert.pem /usr/local/share/ca-certificates/cert.crt",
+                    "update-ca-certificates"
+                ]
+            else:
+                cmds += [
+                    "cp /tmp/cert.pem /etc/pki/ca-trust/source/anchors/cert.pem",
+                    "update-ca-trust"
+                ]
+            self.machine.execute(" && ".join(cmds))
+
+            # Run https server with range option support. QEMU uses range option
+            # see: https://lists.gnu.org/archive/html/qemu-devel/2013-06/msg02661.html
+            # or
+            # https://github.com/qemu/qemu/blob/master/block/curl.c
+            #
+            # and on certain distribution supports only https (not http)
+            # see: block-drv-ro-whitelist option in qemu-kvm.spec for certain distribution
+            server = self.machine.spawn("cd /var/lib/libvirt && python3 /tmp/mock-range-server.py /tmp/server.pem", "httpsserver")
+
+            createTest(TestMachines.VmDialog(self, sourceType='url',
+                                             location=config.ISO_URL,
+                                             memory_size=256, memory_size_unit='MiB',
+                                             storage_size=0, storage_size_unit='MiB',
+                                             os_vendor=config.UNSPECIFIED_VENDOR,
+                                             os_name=config.OTHER_OS,
+                                             start_vm=True))
+
+            # This functionality works on debian only because of extra dep.
+            # Check error is returned if dependency is missing
+            if self.machine.image.startswith("debian"):
+                # remove package
+                self.machine.execute("dpkg -P qemu-block-extra")
+                checkDialogErrorTest(TestMachines.VmDialog(self, sourceType='url',
+                                                           location=config.ISO_URL,
+                                                           memory_size=256, memory_size_unit='MiB',
+                                                           storage_size=0, storage_size_unit='MiB',
+                                                           os_vendor=config.UNSPECIFIED_VENDOR,
```

Comment:
That's a little thin to ensure that the error message makes sense, but :man_shrugging: - it shouldn't happen very often in the first place. Thanks!